azurecompute-arm/src/main/java/org/jclouds/azurecompute/arm/compute/AzureComputeServiceAdapter.java

```diff
@@ -383,39 +391,94 @@ private OSProfile createOsProfile(String computerName, Template template) {
       return builder.build();
    }
 
-   private NetworkInterfaceCard createNetworkInterfaceCard(String subnetId, String name, String locationName,
-                                                           String azureGroup, TemplateOptions options) {
-      final PublicIPAddressApi ipApi = api.getPublicIPAddressApi(azureGroup);
+   private List<NetworkInterfaceCard> createNICs(String nodeName, String location, AzureTemplateOptions options) {
+      ImmutableList.Builder<NetworkInterfaceCard> nics = ImmutableList.builder();
+      String securityGroup = getOnlyElement(options.getGroups(), null);
 
-      PublicIPAddressProperties properties = PublicIPAddressProperties.builder().publicIPAllocationMethod("Static")
-              .idleTimeoutInMinutes(4).build();
+      for (IpOptions ipConfig : publicIpsFirst(options.getIpOptions())) {
```

Comment:
maybe this logic deserves to live in a private method? or a Function name `ipOptionsTo IpConfiguration` maybe?